app-modules/student-data-model/src/Http/Controllers/Api/V1/Students/StudentPrograms/StudentProgramsController.php

```diff
@@ -0,0 +1,91 @@
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\StudentDataModel\Http\Controllers\Api\V1\Students\StudentPrograms;
+
+use AdvisingApp\StudentDataModel\Actions\UpdateStudentProgram;
+use AdvisingApp\StudentDataModel\DataTransferObjects\StudentProgramRequestData;
+use AdvisingApp\StudentDataModel\Http\Resources\Api\V1\StudentProgramResource;
+use AdvisingApp\StudentDataModel\Models\Program;
+use AdvisingApp\StudentDataModel\Models\Student;
+use Dedoc\Scramble\Attributes\Group;
+use Illuminate\Http\Request;
+use Illuminate\Http\Resources\Json\JsonResource;
+use Illuminate\Support\Facades\Gate;
+
+class StudentProgramsController
+{
+    /**
+     * @response StudentProgramResource
+     */
+    #[Group('Students')]
+    public function __invoke(Request $request, UpdateStudentProgram $program, Student $student): JsonResource
+    {
+        Gate::authorize('viewAny', Student::class);
+        Gate::authorize('viewAny', Program::class);
+        Gate::authorize('create', Program::class);
+
+        foreach ($student->programs as $programToUpdate) {
+            Gate::authorize('update', $programToUpdate);
+        }
+
+        foreach ($student->programs as $programToDelete) {
+            Gate::authorize('delete', $programToDelete);
+        }
+
+        $data = $request->validate([
+            'programs' => ['required', 'array'],
+            'programs.*.acad_career' => ['sometimes', 'string', 'max:255'],
+            'programs.*.division' => ['sometimes', 'string', 'max:255'],
+            'programs.*.acad_plan' => ['required', 'array'],
+            'programs.*.prog_status' => ['sometimes', 'string', 'max:255'],
+            'programs.*.cum_gpa' => ['sometimes', 'decimal:0,2'],
+            'programs.*.semester' => ['sometimes', 'string', 'max:255'],
+            'programs.*.descr' => ['sometimes', 'string', 'max:255'],
+            'programs.*.foi' => ['sometimes', 'string', 'max:255'],
+            'programs.*.change_dt' => ['sometimes', 'date', 'date_format:Y-m-d H:i:s'],
+            'programs.*.declare_dt' => ['required', 'date', 'date_format:Y-m-d H:i:s'],
+        ]);
+
+        $studentProgramRequestData = StudentProgramRequestData::from([
+            'programs' => $data['programs'],
+        ]);
+
+        $programs = $program->execute($student, $studentProgramRequestData);
```

Comment:
Like mentioned above you may not need an entire data object just for a collection of programs. You could just make a data collection with the `collect` method on the StudentProgramData DTO.